employee-rostering-frontend/src/ui/components/calendar/EventWrapper.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from "react";
+
+import 'react-big-calendar/lib/css/react-big-calendar.css';
+import './ReactBigCalendarOverrides.css';
+import { Popover } from "@patternfly/react-core";
+
+export default function EventWrapper(props: React.PropsWithChildren<{
+  continuesEarlier: boolean;
+  continuesLater: boolean;
+  className: string;
+  popoverHeader: React.ReactNode;
+  popoverBody: React.ReactNode;
+  style?: React.CSSProperties;
+  additionalProps: {};
+}>): JSX.Element {
+  let className = (props.className)? props.className : "rbc-event";
+  const style: React.CSSProperties = {
+    backgroundColor: "transparent",
+    border: "none",
+    ...props.style
+  };
+
+  if (props.style !== undefined && props.style.top !== undefined && props.style.height !== undefined) {
+    style.gridRowStart = parseInt(props.style.top as string) + 1;
+    style.gridRowEnd = parseInt(props.style.height as string) + style.gridRowStart;
+  
+    if (props.continuesEarlier) {
+      className = className + " continues-from-previous-day";
+    }
+    if (props.continuesLater) {
+      className = className + " continues-next-day";
+    }
+  }
+  
+  return (
+    <Popover
+      className="my-popup"
+      position="right"
+      headerContent={props.popoverHeader}
+      bodyContent={props.popoverBody}
+    >
+      <div
+        className={className}
+        style={style}
+        {...props.additionalProps}
+      >
```

Comment:
Do you mean Popover or popups? Popovers are the things that show when you click on an event (https://www.patternfly.org/v4/documentation/react/components/popover/). Popovers, also known as modals, show up when you are creating a shift. You can customize their color and other properties via CSS variables. There are three built-in styles for Modal (normal, small and large) and one built-in style for Popover.

Reply:
I mean popover. I find the EventWrapper too abstract. I wonder if it would be better to have specialized EventWrappers for different kinds of events (e.g. a ShiftTemplateEventWrapper) to avoid passing `popoverHeader` and `popoverBody` props.

Reply:
`EventWrapper` is a work-around for an issue and create common code that is shared between `Shift`, `Availability` and `ShiftTemplate`. The reason it contains the `Popover` is so that the `Popover` will open when the user click on the time (necessary for short shifts). It hard to move Popover out of there since the Popover must be the top-element of the wrapper for it to capture clicks made on the time.